state.go

```diff
@@ -446,7 +446,10 @@ func (u *UpdateFetchState) Handle(ctx *StateContext, c Controller) (State, bool)
 	}
 
 	// report downloading, don't care about errors
-	c.ReportUpdateStatus(u.update, client.StatusDownloading)
+	merr := c.ReportUpdateStatus(u.update, client.StatusDownloading)
+	if merr != nil && merr.IsFatal() {
+		return NewUpdateErrorState(NewTransientError(merr.Cause()), u.update), false
```

Comment:
I'm a bit fuzzy on the terminology here. The error state is fatal because it will cause an abort of the update, right? But then why do we create a transient error here?